### PR TITLE
batches: more a11y list cleanup

### PR DESCRIPTION
**Pull request by courier-new**
Some "leftovers" that didn't make it into https://github.com/sourcegraph/sourcegraph/pull/44263. These screens aren't currently covered by the third party auditor's testing scope, so I didn't include them in my re-audit last week. However, that doesn't mean they shouldn't receive the same attention eventually. 🙂 This PR:
- Updates the remaining grid lists to use proper semantic elements
- Updates the remaining lists to not announce their headers (and to not skip heading levels)
- Adds additional landmark regions for the pages with these lists
- Updates screen readout for stats on the repo batch changes page
- Removes an outdated comment and an unnecessary prop value

## Test plan

Local testing with screen reader + Storybook stories.

## App preview:

- [Web](https://sg-web-kr-a11y-cleanup.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.28 kb) | 0.01% (+1.28 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/kr/a11y-cleanup/compare-report.html;tab=live_object) for a full comparison between the commits a47e504 and c59b95a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by Piszmog — APPROVED** [review]

File: client/web/src/enterprise/batches/BatchSpecNode.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const BatchSpecNode: React.FunctionComponent<React.PropsWithChildren<Batc
     }, [isExpanded])
 
     return (
-        <>
+        <li className={styles.node}>
```
Comment (eseliger) [inline client/web/src/enterprise/batches/BatchSpecNode.tsx:47]: love it so much

File: client/web/src/enterprise/batches/preview/list/PreviewList.tsx [hunk 2]
```diff
@@ -155,7 +155,6 @@ export const PreviewList: React.FunctionComponent<React.PropsWithChildren<Props>
                 history={history}
                 location={location}
                 useURLQuery={true}
-                listComponent="ul"
```
Comment (eseliger) [inline client/web/src/enterprise/batches/preview/list/PreviewList.tsx:158]: I assume it's the default and that's why you could remove it?

Sourcegraph isn't really of help here, thanks markdown 😆 
Screenshot 2022-11-17 at 01 31 58@2x

**Review by eseliger — APPROVED** [review]

File: client/web/src/enterprise/batches/preview/list/PreviewList.tsx (hunk 2 above)
Comment (courier-new) [reply client/web/src/enterprise/batches/preview/list/PreviewList.tsx:158]: Hahaha helpful. 😂 Yes this is [exactly right](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/components/FilteredConnection/ConnectionNodes.tsx?L124%3A5-124%3A26). 🙂

**Pull request merged**